rockcraft/models/project.py

```diff
@@ -514,13 +514,18 @@ def _add_pebble_data(yaml_data: dict[str, Any]) -> None:
         # Invalid project: let it return to fail in the regular validation flow.
         return
 
-    parts = yaml_data["parts"]
-    if "pebble" in parts:
-        # Project already has a pebble part: this is not supported.
-        raise CraftValidationError('Cannot override the default "pebble" part')
-
     # do not modify the original data with pre-validators
     model = BuildPlanner.unmarshal(copy.deepcopy(yaml_data))
     build_base = model.build_base if model.build_base else model.base
+    pebble_part = Pebble.get_part_spec(build_base)
+
+    parts = yaml_data["parts"]
+    if "pebble" in parts:
+        if parts["pebble"] == pebble_part:
+            # Project already has the correct pebble part.
+            return
+        # Project already has a pebble part, and it's different from ours;
+        # this is currently not supported.
+        raise CraftValidationError('Cannot override the default "pebble" part')
```

Comment:
Thanks for pointing this out. Maybe something like "Cannot change the default pebble part" would be an improvement? Initially this pebble part was an implementation detail that unfortunately leaks out with expand-extensions.

Reply:
I am not sure about how we expect users to use the expand-extension feature and how likely they are to look at the result and think they could modify it. If this is something we do not want the user to play with maybe we should make it explicit that the part is generated and should not be altered? (maybe via a comment in the generated YAML?)
Otherwise I think your suggestion is an improvement but I am not sure it provides enough information for the user to fix the issue (because they would need to figure out what is the default pebble part). Should we print a warning instead and always override the pebble part with the default?

Reply:
The reason the `expand-extensions` command exist is to give the users a way forward in cases where an extension is not directly usable by them. It is reasonable to expect them to update the generated yaml, even if it's not an extremely used feature.

> If this is something we do not want the user to play with maybe we should make it explicit that the part is generated and should not be altered? (maybe via a comment in the generated YAML?)

I like this idea - it would give us some room to revisit this issue in the future. However I don't know how to actually implement this, the yaml generation is just dumping a dict of part properties

Reply:
Thanks.

> I like this idea - it would give us some room to revisit this issue in the future. However I don't know how to actually implement this, the yaml generation is just dumping a dict of part properties

When commenting I was also wondering how to implement that and I am not sure the changes that would be required (change the way we dump the YAML) would be worth it.